backends-common/elasticsearch-v6/src/main/java/org/apache/james/backends/es/v6/ListenerToFuture.java

```diff
@@ -19,21 +19,33 @@
 
 package org.apache.james.backends.es.v6;
 
+import java.util.concurrent.CompletableFuture;
+
 import org.elasticsearch.action.ActionListener;
-import org.elasticsearch.index.reindex.BulkByScrollResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class DeleteByQueryActionListener implements ActionListener<BulkByScrollResponse> {
-    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteByQueryActionListener.class);
+public class ListenerToFuture<T> implements ActionListener<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListenerToFuture.class);
 
-    @Override
-    public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+    private CompletableFuture<T> future;
 
+    public ListenerToFuture() {
+        this.future = new CompletableFuture<>();
```

Comment:
Why do you assign an empty CF?

Reply:
To complete it later ^^

#mutability <3